controllers/discovery/initDiscoveryUserData.js

```diff
@@ -9,11 +9,17 @@ const {QueryTypes} = require('sequelize');
  * @returns
  */
 const InitDiscoveryUserData = async (req, res) => {
-  const {limit = 50, page = 0} = req.body;
+  const {limit = 50, page = 0} = req.query;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from `req.body` to `req.query` is a good one for pagination parameters as these are typically passed in the query string. However, it's important to validate these inputs to prevent SQL injection attacks or unexpected behavior. Consider adding validation for `limit` and `page` to ensure they are integers and within acceptable ranges.
```diff
-  const {limit = 50, page = 0} = req.query;
+  let {limit = 50, page = 0} = req.query;
+  limit = parseInt(limit);
+  page = parseInt(page);
+  if (isNaN(limit) || limit < 1 || limit > MAX_LIMIT) {
+    return res.status(400).json({ error: 'Invalid limit value' });
+  }
+  if (isNaN(page) || page < 0) {
+    return res.status(400).json({ error: 'Invalid page value' });
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->